### ISSUE TITLE
约定式路由 怎么在某个路由禁用全局的layout？

### ISSUE THREAD

**Issue by God-F** [umijs/umi#9895]
例如 全局有一个 layouts/index.tsx 文件做布局  但是在登录页和404页面不需要这个布局  登录页可以根据location信息进行处理 但是404页面应该怎么禁用全局的layouts？

**Comment by fz6m** [comment]
`routes` 配置里用 `layout: false`

**Comment by God-F** [comment]
> `routes` 配置里用 `layout: false`
> 
> 是约定式路由的话，就手写一份 `routes` 配置。
> 
> 如果能控制在某些有限路径下，可以尝试用 [useMatch](https://umijs.org/docs/api/api#usematch) 在 layout 里判断下。

是的 我现在的情况就是 目前是约定式路由 登录页面我完全可以在layout中根据路由信息进行判断  但是404页面好像没有办法 进行处理 所以现在是不是只能用配置式路由了？

**Comment by fz6m** [comment]
配置式路由只需要配置 `layout: false` 就行了。

注意看我上面说的，你可以在 layout 里用 `useMatch` 拿到当前路由信息，如果你可以确定某些范围，就可以选择性控制是否渲染 layout 。

**Comment by livelybone** [comment]
之前 umi3 也支持约定式路由设置 layout: false 的，大概是：

**Comment by fz6m** [comment]
有一个 PR （ #10593 ）可以支持约定式路由关闭 layout ，但是很久了，代码有很多变化。要参考着新开一个 PR 才行。
